scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -231,49 +232,52 @@ def _str2bool(v: str) -> bool:
     logger.print(f"create list: {publish_list}")
 
     failure_list = []
-    for asset in util.find_assets(input_dirs=assets_dir):
-        asset_names = publish_list.get(asset.type.value, [])
-        if not ("*" in asset_names or asset.name in asset_names):
-            logger.print(
-                f"Skipping registering asset {asset.name} because it is not in the publish list")
+    for publish_asset_type in PUBLISH_ORDER:
+        if publish_asset_type.value not in publish_list:
             continue
-        logger.print(f"Registering {asset}")
-        final_version = asset.version + "-" + \
-            passed_version if passed_version else asset.version
-        logger.print(f"final version: {final_version}")
-        asset_ids[asset.name] = ASSET_ID_TEMPLATE.substitute(
-            registry_name=registry_name,
-            asset_type=f"{asset.type.value}s",
-            asset_name=asset.name,
-            version=final_version,
-        )
+        for asset in util.find_assets(input_dirs=assets_dir, types=[publish_asset_type]):
```

Comment:
This is one of the more expensive operations. You probably want to do this first for all, without specifying `types=...` and then have inner loops for PUBLISH_ORDER, then publish_list.

Reply:
Made changes to this.